kore/src/Kore/ASTVerifier/SentenceVerifier.hs

```diff
@@ -375,12 +392,30 @@ verifyClaimSentence sentence =
         attrs <-
             parseAndVerifyAxiomAttributes verifiedModule'
             $ sentenceClaimAttributes sentence
+        when (rejectClaim attrs verified)
+            $ sentenceClaimError (SentenceClaim verified)
         State.modify' $ addClaim (SentenceClaim verified) attrs
   where
     addClaim verified attrs =
         Lens.over
             (field @"indexedModuleClaims")
             ((attrs, verified) :)
+    rejectClaim attrs verified =
+        case fromSentenceAxiom (attrs, verified) of
+            Right (OnePathClaimPattern (OnePathRule rulePattern))
+              | rejectRulePattern rulePattern -> True
+            Right (AllPathClaimPattern (AllPathRule rulePattern))
+              | rejectRulePattern rulePattern -> True
+            _ -> False
+    rejectRulePattern
+        RulePattern { left, antiLeft, requires, rhs = RHS { right, ensures } }
+      =
+        (not . any containsForall)
+            (catMaybes [antiLeft] <> [left, unwrapPredicate requires])
+        && any containsForall [right, unwrapPredicate ensures]
+    sentenceClaimError sentenceClaim =
+        withSentenceClaimContext sentenceClaim $ koreFail "Found claim with \
```

Comment:
I was partly mistaken: we already use `withSentenceClaimContext` above, so we don't need to repeat it here.